db/document.go

```diff
@@ -332,6 +333,19 @@ func (doc *Document) GetMetaMap(userXattrKey string) (map[string]interface{}, er
 	}, nil
 }
 
+func (doc *Document) SetCrc32cUserXattrHash(userXattrKey string) {
+	if len(doc.rawUserXattr) > 0 {
+		doc.SyncData.Crc32cUserXattr = base.Crc32cHashString(doc.rawUserXattr)
+		return
+	}
+
+	// If no key is defined but there is a value in its crc32 then we should update the crc32 to be empty. This would
+	// happen if the feature was enabled but later disabled.
+	if userXattrKey == "" && doc.SyncData.Crc32cUserXattr != "" {
+		doc.SyncData.Crc32cUserXattr = ""
+	}
+}
```

Comment:
I think this might be the same comment as Adam's earlier - but if there was a user xattr value on a doc and then it got removed, but the user still has the `userXattrKey` set (e.g. for other docs), won't that keep around the old crc value indefinitely?

I.e. Should we also be emptying the value when `len(rawUserXattr) == 0` ?